synapse/lib/stormlib/log.py

```diff
@@ -0,0 +1,168 @@
+            Log a message to the Cortex at the warning log level.
+
+            Notes:
+                This requires the ``storm.lib.log.warning`` permission to use.
+
+            Examples:
+                Log a warning message::
+
+                    $lib.log.warning('I am a warning message!')
+
+                Log a warning message with extra information::
+
+                    $lib.log.warning('Extra information included here.', extra=$lib.dict(key=$valu))''',
+         'type': {'type': 'function', '_funcname': '_logDebug',
+                  'args': (
+                      {'name': 'mesg', 'type': 'str', 'desc': 'The message to log.'},
+                      {'name': 'extra', 'type': 'dict', 'desc': 'Extra key / value pairs to include when structured '
+                                                                'logging is enabled on the Cortex.',
+                       'default': None, },
+                  ),
+                  'returns': {'type': 'null'},
+                  }
+         },
+        {'name': 'error', 'desc': '''
+            Log a message to the Cortex at the error log level.
+
+            Notes:
+                This requires the ``storm.lib.log.error`` permission to use.
+
+            Examples:
+                Log a error message::
+
+                    $lib.log.error('I am a error message!')
+
+                Log a error message with extra information::
+
+                    $lib.log.error('Extra information included here.', extra=$lib.dict(key=$valu))''',
+         'type': {'type': 'function', '_funcname': '_logDebug',
+                  'args': (
+                      {'name': 'mesg', 'type': 'str', 'desc': 'The message to log.'},
+                      {'name': 'extra', 'type': 'dict', 'desc': 'Extra key / value pairs to include when structured '
+                                                                'logging is enabled on the Cortex.',
+                       'default': None, },
+                  ),
+                  'returns': {'type': 'null'},
+                  }
+         },
+    )
+
+    _storm_lib_path = ('log',)
+
+    def getObjLocals(self):
+        return {
+            'info': self._logInfo,
+            'debug': self._logDebug,
+            'error': self._logError,
+            'warning': self._logWarning,
+        }
+
+    async def _getExtra(self, extra=None):
+        if extra is None:
+            return extra
+        extra = await s_stormtypes.toprim(extra)
+        if extra and not isinstance(extra, dict):
+            mesg = f'extra provided to log call must be a dictionary compatible type. Got {extra.__class__.__name__} ' \
+                   f'instead.'
+            raise s_exc.BadArg(mesg=mesg, arg='extra')
+        extra = {'synapse': extra}
+        return extra
+
+    @s_stormtypes.stormfunc(readonly=True)
+    async def _logDebug(self, mesg, extra=None):
+        self.runt.confirm(('storm', 'lib', 'log', 'debug'), gateiden=None)
```

Comment:
dont specify gateiden=None when this is already the default

Reply:
fixed